src/main/java/de/tum/in/www1/artemis/web/rest/TutorEffortResource.java

```diff
@@ -0,0 +1,164 @@
+                TutorEffort effort = createTutorEffortWithInformation(currentUserId, currentUserEvents, submissionsPerTutor.get(currentUserId));
+                tutorEffortList.add(effort);
+            }
+        });
+        return ResponseEntity.ok().body(tutorEffortList);
+    }
+
+    /**
+     * Takes in parameters and sets respective properties on a new TutorEffort object
+     * @param userId the id of the user to set
+     * @param events the events of the respective user
+     * @param submissions the number of submissions the tutor assessed
+     * @return a TutorEffort object with all the data set
+     */
+    private TutorEffort createTutorEffortWithInformation(Long userId, List<TextAssessmentEvent> events, int submissions) {
+        TutorEffort effort = new TutorEffort();
+        effort.setUserId(userId);
+        effort.setCourseId(events.get(0).getCourseId());
+        effort.setExerciseId(events.get(0).getTextExerciseId());
+        effort.setTotalTimeSpentMinutes(calculateTutorOverallTimeSpent(events));
+        effort.setNumberOfSubmissionsAssessed(submissions);
+        return effort;
+    }
+
+    /**
+     * Traverses over tutorEvents (which contain timestamps). Calculates time spent between timestamps
+     * assuming timestamps are not further than THRESHOLD_MINUTES away from each other.
+     * @param tutorEvents events to be analysed
+     * @return the number of minutes spent
+     */
+    private int calculateTutorOverallTimeSpent(List<TextAssessmentEvent> tutorEvents) {
+        int timeSeconds = 0;
+        int index = 0;
+        while (index < tutorEvents.size()) {
+            TextAssessmentEvent current = tutorEvents.get(index);
+            try {
+                // access next element & catch out of bounds if end of the list.
+                TextAssessmentEvent next = tutorEvents.get(index + 1);
+                int diffInSeconds = getDateDiffInSeconds(Date.from(current.getTimestamp()), Date.from(next.getTimestamp()));
+                if (diffInSeconds <= THRESHOLD_MINUTES * 60) {
+                    timeSeconds += diffInSeconds;
+                }
+            }
+            catch (IndexOutOfBoundsException ignored) {
+            }
+            index++;
+        }
+        return timeSeconds / 60;
+    }
+
+    /**
+     * Groups assessment events by user id into a map
+     * @param events the events to query from
+     * @return a map with key representing user id and value representing respective list of events for the user
+     */
+    private Map<Long, List<TextAssessmentEvent>> groupByUserId(List<TextAssessmentEvent> events) {
+        Map<Long, List<TextAssessmentEvent>> map = new HashMap<>();
+
+        events.forEach((event) -> {
+            Long cUserId = event.getUserId();
+            var cEvents = map.get(cUserId);
+            // key, value pair doesn't exist, initialize empty list
+            if (cEvents == null) {
+                cEvents = new ArrayList<>();
+            }
```

Comment:
Should we better use `getOrDefault` instead?
```suggestion
            var cEvents = map.getOrDefault(cUserId, new ArrayList<>());
```